docs/cms-query-integration.md

```diff
@@ -0,0 +1,68 @@
+# CMS Query Integration
+
+The Search module overrides the default search functionality to query the specialized search index in place of a standard MySQL query. This means all default search operations using the CMS search APIs for posts, users and terms such as `WP_Query`, `WP_User_Query`, `WP_Term_Query` and the REST API search endpoints will transparently make use of the search index by default.
+
+It is possible to control this behaviour and thus use the search index for non search queries that might otherwise be too slow to do in the database such as meta queries.
+
+Each query class accepts an array of parameters. ElasticPress checks for the value of an `ep_integrate` parameter, if this is true the query is performed by Elasticsearch instead of MySQL. If the `s` (search) parameter is present `ep_integrate` will be set to `true` automatically.
+
+The following example sets `ep_integrate` to true for a non search query, that is any query where the `s` parameter is not used.
+
+```php
+$posts = new WP_Query( [
+	'ep_integrate' => true, // Use Elasticsearch.
+	'post_type' => 'project',
+	'meta_query' => [
+		[
+			'key' => 'color',
+			'value' => [ 'blue', 'green' ],
+			'compare' => 'IN'
+		],
+	],
+] );
+```
+
+Note in the above example you would need a PHP Code Sniffer exception for using the meta query to pass [Automated Code Review](docs://guides/code-review/README.md).
```

Comment:
Comma after "in the above example." Also, if a PHPCS ignore is required, why not add it to the example (as has been done in other docs)? And then explain why the ignore is there instead.

```suggestion
Note in the above example, you would need a PHP Code Sniffer exception for using the meta query to pass [Automated Code Review](docs://guides/code-review/README.md).
```

Reply:
Not sure this one is necessary myself but have noticed that I've been a bit lazy with that sentence.
```suggestion
Note that in the above example you would need a PHP Code Sniffer exception for using the meta query to pass [Automated Code Review](docs://guides/code-review/README.md).
```

Reply:
Good point re. adding the phpcs rule too. I'll let into what that is.